service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/gatekeepingorder/AddGatekeepingOrderControllerAboutToSubmitTest.java

```diff
@@ -0,0 +1,167 @@
+import uk.gov.hmcts.reform.fpl.model.event.GatekeepingOrderEventData;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+import uk.gov.hmcts.reform.fpl.service.docmosis.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.fpl.Constants.LOCAL_AUTHORITY_1_CODE;
+import static uk.gov.hmcts.reform.fpl.enums.OrderStatus.DRAFT;
+import static uk.gov.hmcts.reform.fpl.enums.OrderStatus.SEALED;
+import static uk.gov.hmcts.reform.fpl.enums.OrderType.CARE_ORDER;
+import static uk.gov.hmcts.reform.fpl.enums.State.CASE_MANAGEMENT;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocument;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocumentBinaries;
+
+@WebMvcTest(AddGatekeepingOrderController.class)
+@OverrideAutoConfiguration(enabled = true)
+class AddGatekeepingOrderControllerAboutToSubmitTest extends AbstractCallbackTest {
+    AddGatekeepingOrderControllerAboutToSubmitTest() {
+        super("add-gatekeeping-order");
+    }
+
+    private static final Document SDO_DOCUMENT = testDocument();
+    private static final Document C6_DOCUMENT = testDocument();
+    private static final DocumentReference SDO_REFERENCE = DocumentReference.buildFromDocument(SDO_DOCUMENT);
+    private static final DocumentReference C6_REFERENCE = DocumentReference.buildFromDocument(C6_DOCUMENT);
+
+    @MockBean
+    private DocmosisDocumentGeneratorService documentGeneratorService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    private final Time time = new FixedTimeConfiguration().fixedDateTime(LocalDateTime.of(2021, 3, 3, 0, 0, 0));
+
+    @BeforeEach
+    void setup() {
+        final byte[] sdoBinaries = testDocumentBinaries();
+        final byte[] c6Binaries = testDocumentBinaries();
+        final String sealedOrderFileName = "standard-directions-order.pdf";
+        final String c6FileName = "c6.pdf";
+
+        given(documentGeneratorService.generateDocmosisDocument(any(DocmosisStandardDirectionOrder.class), any()))
+            .willReturn(new DocmosisDocument(sealedOrderFileName, sdoBinaries));
+
+        given(documentGeneratorService.generateDocmosisDocument(any(DocmosisNoticeOfProceeding.class), any()))
+            .willReturn(new DocmosisDocument(c6FileName, c6Binaries));
+
+        given(uploadDocumentService.uploadPDF(sdoBinaries, sealedOrderFileName)).willReturn(SDO_DOCUMENT);
+        given(uploadDocumentService.uploadPDF(c6Binaries, c6FileName)).willReturn(C6_DOCUMENT);
+    }
+
+    @Test
+    void shouldBuildDraftSDOWithExistingDraftDocumentWhenOrderStatusIsDraft() {
```

Comment:
test explicitly if order status is null (i.e forced to be a draft)

Reply:
service.buildBaseGatekeepingOrder(caseData) sets status to DRAFT if null so this won't happen in the callback